File: backend/src/users/reachability-categories.ts
/** The fixed set of things a Reachable user can say they're open to being
 * approached about — matches the product spec's "Reachability Preferences"
 * checkboxes. Kept as plain keys (not free text) so Ask HERE / Ping matching
 * can filter on them later; the human-readable label lives client-side. */
export const REACHABILITY_CATEGORIES = [
  'local_questions',
  'recommendations',
  'travel',
  'technology',
  'professional_advice',
  'study',
  'hobbies',
  'helping',
  'conversation',
  'friends',
  'dating',
] as const;

export type ReachabilityCategory = (typeof REACHABILITY_CATEGORIES)[number];

File: app/lib/data/reachability_categories.dart
/// The fixed set of things a Reachable user can say they're open to being
/// approached about. Keys must match the backend's REACHABILITY_CATEGORIES
/// exactly — labels are English source text, run through `t()` at display
/// time like every other UI string.
class ReachabilityCategory {
  const ReachabilityCategory(this.key, this.label);

  final String key;
  final String label;
}

const reachabilityCategories = [
  ReachabilityCategory('local_questions', 'Local questions'),
  ReachabilityCategory('recommendations', 'Recommendations'),
  ReachabilityCategory('travel', 'Travel'),
  ReachabilityCategory('technology', 'Technology'),
  ReachabilityCategory('professional_advice', 'Professional advice'),
  ReachabilityCategory('study', 'Study'),
  ReachabilityCategory('hobbies', 'Hobbies'),
  ReachabilityCategory('helping', 'Helping someone'),
  ReachabilityCategory('conversation', 'Conversation'),
  ReachabilityCategory('friends', 'Making friends'),
  ReachabilityCategory('dating', 'Dating / Meeting someone'),
];

String labelForCategory(String key) =>
    reachabilityCategories.firstWhere((c) => c.key == key, orElse: () => ReachabilityCategory(key, key)).label;
